File: Java/프로그래머스/2/12913. 땅따먹기/땅따먹기.java
class Solution {
    int solution(int[][] land) {
        int answer = 0;
        
        for (int i = 1; i<land.length; i++) {
            for (int j = 0; j<land[i].length; j++) {
                int max = 0;
                for (int x = 0; x<land[i].length; x++) {
                    if (x == j) continue;
                    max = Math.max(max, land[i-1][x]);
                }
                land[i][j] += max;
            }
        }

        int lastRow = land.length-1;
        for (int i = 0; i<4; i++) {
            answer = Math.max(answer, land[lastRow][i]);
        }
        return answer;
    }
}

File: Swift/프로그래머스/2/12913. 땅따먹기/땅따먹기.swift
import Foundation

func solution(_ land:[[Int]]) -> Int{
    var dp: [[Int]] = [land[0]]
    for i in 1..<land.count {
        var row: [Int] = []
        for j in 0..<4 {
            var previousMax = 0
            for k in 0..<4 {
                guard k != j else { continue }
                previousMax = max(previousMax, dp[i-1][k])
            }
            row.append(previousMax + land[i][j])
        }
        dp.append(row)
    }
    return dp.last!.max()!
}